Set4Feedback/set4-draft-rated/28/line/next.cc

```diff
@@ -0,0 +1,19 @@
+#include "line.ih"
+
+string Line::next()
+{
+    if (d_position == string::npos)
+        return "";
+                                                        // Find next ws 
+    size_t endPosition = d_line.find_first_of(" \t", d_position);
+    
+    string subString =                                  // Cut next substring
+        (endPosition == string::npos)
+        ? d_line.substr(d_position)
+        : d_line.substr(d_position, endPosition - d_position);
+        // HB: you can simplify so you don't need the ternary
```

Comment:
```suggestion
    string const subString = d_line.substr(d_position, endPosition - d_position);
```

I think this is already fine. `string.substr` only consumes up to whatever is available, even if `endPosition` is equal to `string:::npos`. There's also no danger with `endPosition` wrapping around `string::npos`.

Reply:
Yeah that's what we had here initially! This does help with clarity a bit for the action itself, but it makes the line as a whole a bit more clunky

Reply:
I kept it in this one, but have amended it in the working directory! Wouldn't mind keeping the erroneous/outdated stuff for reference and posterity